test/libsolutil/CommonIO.cpp

```diff
@@ -98,6 +118,426 @@ BOOST_AUTO_TEST_CASE(readBytes_past_end)
+	BOOST_TEST(absolutePath("./a/", reference) == absolutePathJoin(parent, "a"));
+	BOOST_TEST(absolutePath("./a/b", reference) == absolutePathJoin(parent, "a/b"));
+	BOOST_TEST(absolutePath("./a/b/", reference) == absolutePathJoin(parent, "a/b"));
+
+	BOOST_TEST(absolutePath("./@", reference) == absolutePathJoin(parent, "@"));
+	BOOST_TEST(absolutePath("./:", reference) == absolutePathJoin(parent, ":"));
+	BOOST_TEST(absolutePath("./|", reference) == absolutePathJoin(parent, "|"));
+	BOOST_TEST(absolutePath("./<>", reference) == absolutePathJoin(parent, "<>"));
+	BOOST_TEST(absolutePath("./123", reference) == absolutePathJoin(parent, "123"));
+	BOOST_TEST(absolutePath("./https://example.com", reference) == absolutePathJoin(parent, "https:/example.com"));
+
+	BOOST_TEST(absolutePath(".//a", reference) == absolutePathJoin(parent, "a"));
+	BOOST_TEST(absolutePath(".//a//", reference) == absolutePathJoin(parent, "a"));
+	BOOST_TEST(absolutePath(".//a//b", reference) == absolutePathJoin(parent, "a/b"));
+	BOOST_TEST(absolutePath(".//a//b//", reference) == absolutePathJoin(parent, "a/b"));
+
+	BOOST_TEST(absolutePath("./a\\", reference) == absolutePathJoin(parent, "a\\"));
+	BOOST_TEST(absolutePath("./a\\b", reference) == absolutePathJoin(parent, "a\\b"));
+	BOOST_TEST(absolutePath("./a\\b\\", reference) == absolutePathJoin(parent, "a\\b\\"));
+
+	BOOST_TEST(absolutePath("./a/.", reference) == absolutePathJoin(parent, "a"));
+	BOOST_TEST(absolutePath("./a/b/.", reference) == absolutePathJoin(parent, "a/b"));
+	BOOST_TEST(absolutePath("./a/./b/.", reference) == absolutePathJoin(parent, "a/b"));
+
+	BOOST_TEST(absolutePath("./a/..", reference) == absolutePathJoin(parent, ""));
+	BOOST_TEST(absolutePath("./a/../", reference) == absolutePathJoin(parent, ""));
+	BOOST_TEST(absolutePath("./a/b/..", reference) == absolutePathJoin(parent, "a"));
+	BOOST_TEST(absolutePath("./a/b/../", reference) == absolutePathJoin(parent, "a"));
+	BOOST_TEST(absolutePath("./a/../b", reference) == absolutePathJoin(parent, "b"));
+	BOOST_TEST(absolutePath("./a/../b/", reference) == absolutePathJoin(parent, "b"));
+	BOOST_TEST(absolutePath("./a/../b/..", reference) == absolutePathJoin(parent, ""));
+	BOOST_TEST(absolutePath("./a/../b/../", reference) == absolutePathJoin(parent, ""));
+
+	BOOST_TEST(absolutePath("./a\\..", reference) == absolutePathJoin(parent, "a\\.."));
+	BOOST_TEST(absolutePath("./a\\..\\", reference) == absolutePathJoin(parent, "a\\..\\"));
+	BOOST_TEST(absolutePath("./a\\b\\..", reference) == absolutePathJoin(parent, "a\\b\\.."));
+	BOOST_TEST(absolutePath("./a\\b\\..\\", reference) == absolutePathJoin(parent, "a\\b\\..\\"));
+	BOOST_TEST(absolutePath("./a\\..\\b", reference) == absolutePathJoin(parent, "a\\..\\b"));
+	BOOST_TEST(absolutePath("./a\\..\\b\\", reference) == absolutePathJoin(parent, "a\\..\\b\\"));
+	BOOST_TEST(absolutePath("./a\\..\\b\\..\\", reference) == absolutePathJoin(parent, "a\\..\\b\\..\\"));
```

Comment:
Yep, paths with backslashes and/or double slashes behave differently on Windows.